pkg/get/tools.go

```diff
@@ -4255,5 +4266,67 @@ https://github.com/{{.Owner}}/{{.Repo}}/releases/download/{{.Version}}/{{.Repo}}
 
 			{{.Name}}_{{.VersionNumber}}_{{$osStr}}_{{$arch}}.{{$ext}}`,
 		})
+
+	tools = append(tools,
+		Tool{
+			Owner:       "go",
+			Repo:        "go",
+			Name:        "go",
+			SystemOnly:  true,
+			Version:     "1.22.4",
```

Comment:
Since go does not uses github or k8s version strategy, will it be correct to hard-code it?

Reply:
One option is to hard-code it, the other thing we could have a strategy for go and move the code from the system installer there. The version can already be determined.

@rgee0 what are your thoughts? Go is installed via `system install` with custom code, we are looking at using the template system instead.

Reply:
I was thinking to execute go version detection code to be only executed when cli is actually installing the go. I am not sure, how to setup that in current code base.

Reply:
I'd suggest a similar approach to the exiting version strategies but using: https://go.dev/VERSION?m=text

Reply:
> I'd suggest a similar approach to the exiting version strategies but using: https://go.dev/VERSION?m=text

This is implemented in system [install](https://github.com/alexellis/arkade/blob/master/cmd/system/go.go#L111-L142) but we are stuck around how to implement same in template approach

Reply:
Apologies I missed `The version can already be determined`, maybe im also missing some context, but you're using `GetDownloadURL` which will take you into `GetUrl` which is where the existing version strategies are implemented.  `tool` is available all the way through so a `goVersionStrategy` could be implemented in there in much the same way as k8s is.